src/lib.rs

```diff
@@ -1479,6 +1565,59 @@ impl<'a> MatchTreeStep<'a> {
                     }
                 }
             }
+            TypedFormat::RepeatBetween(_, xmin, xmax, a) => {
+                let min_bounds = xmin.bounds();
+                let max_bounds = xmax.bounds();
+                match (min_bounds.is_exact(), max_bounds.is_exact()) {
+                    (Some(min), Some(max)) => match min.cmp(&max) {
+                        Ordering::Less => {
+                            if min > 0 {
+                                Self::from_gt_format(
+                                    module,
+                                    &**a,
+                                    Rc::new(Next::RepeatBetween(
+                                        min - 1,
+                                        max - 1,
+                                        MaybeTyped::Typed(&**a),
+                                        next,
+                                    )),
+                                )
+                            } else {
+                                Self::from_next(
+                                    module,
+                                    Rc::new(Next::RepeatMax(
+                                        max,
+                                        MaybeTyped::Typed(&**a),
+                                        next.clone(),
+                                    )),
+                                )
+                            }
+                        }
+                        Ordering::Equal => {
+                            let next = next.clone();
+                            if min > 0 {
+                                Self::from_gt_format(
+                                    module,
+                                    &**a,
+                                    Rc::new(Next::RepeatCount(
+                                        min - 1,
+                                        MaybeTyped::Typed(&**a),
+                                        next,
+                                    )),
+                                )
+                            } else {
+                                Self::from_next(module, next)
+                            }
+                        }
+                        Ordering::Greater => {
+                            panic!("incoherent repeat-between: min {} > max {}", min, max)
+                        }
+                    },
+                    _ => {
+                        unreachable!("inexact repeat-between bounds (not technically a problem but not what the combinator was designed for...");
```

Comment:
Possibly better to allow variable/dependent bounds?

Reply:
I think we just shrug and cut off the match tree, much as we do for e.g. `RepeatUntilLast` and `RepeatUntilSeq`; just means you can't switch on something that starts with a `RepeatBetween` of unknown size, which is fine, it's still a legal format to have in cases where you don't need to switch on it.

Reply:
I left a comment on (#168) to revisit these lines, hopefully nothing should break in the meantime so there oughtn't be a pressing need to change this right away. If necessary, we can always return to this. Will mentally bookmark this as well.